source_files/gpu.c

```diff
@@ -38,98 +38,103 @@
     if ( cuerr != cudaSuccess)
     {
-        fprintf( stderr, "    - CUDA_SAFE_CALL() failed at %s:%i : %i %s\n",
-                 file, line, cuerr,cudaGetErrorString( cuerr ) );
+        fprintf( stderr, "    - CUDA error at %s:%i : error code = %i(%s), device index:%i\n",
+                 file, line, cuerr, cudaGetErrorString( cuerr ), dev_index );
         exit(cuerr);
     }
-#endif
 
     return;
 }
 
-int roundUp(int numToRound, int multiple)  
-{  
- if(multiple == 0)  
- {  
-  return numToRound;  
- }  
+static int roundUp(int numToRound, int multiple) {  
+    if(multiple == 0) {  
+        return numToRound;
+    }  
 
- int remainder = numToRound % multiple; 
- if (remainder == 0)
-  {
-    return numToRound; 
-  }
+    int remainder = numToRound % multiple; 
+    if (remainder == 0) {
+        return numToRound;
+    }
 
- return numToRound + multiple - remainder; 
+    return numToRound + multiple - remainder; 
 } 
 
-int ipow(int base,int exp) {
+static int ipow(int base,int exp) {
     int result = 1;
     while (exp) {
         if (exp & 1) result *= base;
         exp >>= 1;
         base *= base;
     }
+
     return result;
 }
 
 static int bbs(void) {
     static int s;
     static int runned = 0;
+
     int n = 131*17;
     if(runned == 0) {
         runned = 1;
         s = 123; // this is the seed
     }
     s = ipow(s,2) % n;
+
     return s;
 }
 
-void* fillup(void* array, int useD, int size) {
+static void* fillup(void* array, int useD, int size) {
     int i;
+    
     if(useD) {
         double frac;
         double *dbl=(double*)array;
         dbl = malloc(sizeof(double)*size*size);
+        if(!dbl) {
+            fprintf(stderr, "Could not allocate memory for GPU computation\n");
+            exit(ENOMEM);
+        }
         for (i=0; i<size*size; i++) {
-            if(i % 128 == 0) {
+            if(i % 128 == 0) {   
                 frac = (double) bbs() / 10000;
                 dbl[i]=(double) bbs() + frac;
             }
         }
-    } else {
+    }
+    else {
         float frac;
         float *flt = (float*)array;
         flt = malloc(sizeof(float)*size*size);
+        if(!flt) { 
+            fprintf(stderr, "Could not allocate memory for GPU computation\n");
+            exit(ENOMEM);   
+        }
         for (i=0; i<size*size; i++) {
             if(i % 128 == 0) {
                 frac = (float) bbs() / 10000;
                 flt[i]=(float) bbs() + frac;
             }
         }
     }
+
     return array;
 }
 void* startBurn(void *index) {
     int d_devIndex = *((int*)index);   //GPU Index. Used to pin this pthread to the GPU.
     int d_iters,i;
     int pthread_useDouble = useDouble; //local per-thread variable, if there's a GPU in the system without Double Precision support.
     int size_use=0;
-    if (globsize>0){
-        size_use=globsize;
+    if (gpuvar->msize>0){
+        size_use=gpuvar->msize>0;
```

Comment:
Ôo size_use=gpuvar->msize>0; are you sure, you want to set it to the result of a comparison